src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpOnTypeFormattingPass.cs

```diff
@@ -272,16 +272,6 @@ private static void CleanupSourceMappingStart(FormattingContext context, Range s
 
             if (newLineCount == 0)
             {
-                // If on type formatting is happening on a single line then we just clean up the start to one space
-                // so @{    throw null; } will be formatted to @{ throw null; }
-                // Ideally we'd put that across three lines, which is what normal formatting does, but since we
-                // can't control the cursor, that doesn't end well.
-                if (isOnType && sourceMappingRange.Start.Line == sourceMappingRange.End.Line)
```

Comment:
So now the logic is that we're opinionated and say that it should always be multi-line. Is a good summation of our original goals / where we ended up that we tried formatting it to a single line and found that there were other issues with that and are now falling back to mulit-line formatting?

Reply:
Not quite.. Previously it would format to two lines instead of three, so the original intention was to fix that. Because of cursor issues with the platform, doing that was a bad user experience, so we compromised and said "if its on one, we'll leave it there because we can't deliver a better experience". Now we can deliver a better experience, so we are.

We're already opinionated and say this should be over 3 lines, but only in whole document formatting, not in on-type.

Reply:
Ahhh ok cool, thanks for clarifying. That makes sense!